themes/doom-rouge-theme.el

```diff
@@ -0,0 +1,182 @@
+;;; doom-rouge-theme.el --- inspired by Nord -*- no-byte-compile: t; -*-
+(require 'doom-themes)
+
+;;
+(defgroup doom-rouge-theme nil
+  "Options for doom-themes"
+  :group 'doom-themes)
+
+(defcustom doom-rouge-brighter-comments nil
+  "If non-nil, comments will be highlighted in more vivid colors."
+  :group 'doom-rouge-theme
+  :type 'boolean)
+
+(defcustom doom-rouge-brighter-tabs t
+  "If non-nil, tabs will a more vivid background color."
+  :group 'doom-rouge-theme
+  :type 'boolean)
+
+(defcustom doom-rouge-comment-bg doom-rouge-brighter-comments
+  "If non-nil, comments will have a subtle, darker background. Enhancing their
+legibility."
+  :group 'doom-rouge-theme
+  :type 'boolean)
+
+(defcustom doom-rouge-padded-modeline doom-themes-padded-modeline
+  "If non-nil, adds a 4px padding to the mode-line. Can be an integer to
+determine the exact padding."
+  :group 'doom-rouge-theme
+  :type '(choice integer boolean))
+
+;;
+(def-doom-theme doom-rouge
+  "A dark theme inspired by Nord."
+
+  ;; name        default   256       16
+  ((bg         '("#172030" nil       nil            )) ;; modified
+   (bg-alt     '("#172030" nil       nil            ))
+   (base0      '("#070A0E" "black"   "black"        ))
+   (base1      '("#0E131D" "#1e1e1e" "brightblack"  ))
+   (base2      '("#151D2B" "#2e2e2e" "brightblack"  ))
+   (base3      '("#172030" "#262626" "brightblack"  ))
+   (base4      '("#5D636E" "#3f3f3f" "brightblack"  ))
+   (base5      '("#64727d" "#64727d" "brightblack"  ))
+   (base6      '("#B16E75" "#6b6b6b" "brightblack"  ))
+   (base7      '("#E8E9EB" "#979797" "brightblack"  ))
+   (base8      '("#F0F4FC" "#dfdfdf" "white"        ))
+   (fg         '("#BBBBBB"    "#bbb"    "white"        ))
+   (fg-alt     '("#E5E9F0" "#bfbfbf" "brightwhite"  ))
+
+   (grey       base5)
+   (red        '("#c6797e" "#c6797e" "red"          ))
+   (light-red  '("#DB6E8F" "#DB6E8F" "red"          ))
+   (orange     '("#eabe9a" "#eabe9a" "brightred"    ))
+   (green      '("#A3B09A" "#A3B9A4" "green"        ))
+   (teal       '("#7ea9a9" "#7ea9a9" "brightgreen"  ))
+   (yellow     '("#F7E3AF" "#F7E3AF" "yellow"       ))
+   (blue       '("#6e94b9" "#6e94b9" "brightblue"   ))
+   (dark-blue  '("#1E6378" "#1E6378" "blue"         ))
+   (magenta    '("#b18bb1" "#b18bb1" "magenta"      ))
+   (salmon     '("#F9B5AC" "#F9B5AC" "orange"       ))
+   (violet     '("#5D80AE" "#5D80AE" "brightmagenta"))
+   (cyan       '("#88C0D0" "#88C0D0" "brightcyan"   ))
+   (dark-cyan  '("#507681" "#507681" "cyan"         ))
+
+   ;; face categories -- required for all themes
+   (highlight      base6)
+   (vertical-bar   (doom-darken base1 0.2))
+   (selection      base4)
+   (builtin        light-red)
+   (comments       grey)
+   (doc-comments   green)
+   (constants      red)
+   (functions      salmon)
+   (keywords       magenta)
+   (methods        light-red)
+   (operators      magenta)
+   (type           magenta)
```

Comment:
This is something that I explicitly made different. This is due to the difference in markup of code from VSCode to Emacs. This might be something that just needs to be overwritten on a language bases. But this significantly shifts from what it looks like for Ruby when comparing the themes side by side

Reply:
Not sure if we want to go down that route but in keeping with the 'spirit' of the theme I think keeping type faces red reflects more closely to the theme in VS Code

Reply:
The thing is, the theme was designed specifically for web dev (JS, HTML, CSS). If you check the repo, the only language specific customization is for those languages. All the other (C, Python, Ruby...) fall into the defaults I think, so what I did was to grab specifically the JS colors. By the way, you now may try centaur-tabs plain icons custom to make the theme even better.

Reply:
Do you have a code example where this is diverging for js/html when type is set back to red/orange compared to magenta. VSCode technically uses both the red/orange/yellow to highlight different identifiers/types in the code. But in emacs we lose most of that. Setting the type to red brought more of it into the theme. Copying exact matches for specific kinds of 'faces' from VSCode might be technically right but it looks significantly different when comparing the two side  by side.

Most of the improvements on color choice here are great. This is is just something that I think we will need to make our own for the conversion to an emacs theme. We lose out on a ton of different coloring due to the difference in syntax highlighting. And the red is the center piece of the whole theme.

Reply:
I would need to look into it. Could you show me an screenshot contrasting the two options in order to be in the same page?

Reply:
Here is JS example. Javascript looks nothing like the vscode theme because of the lack of face modification for normal variables.

![image](https://user-images.githubusercontent.com/1653499/76777363-16877780-6776-11ea-9867-f1af11c6d837.png)

Here is ruby with the change you made
![image](https://user-images.githubusercontent.com/1653499/76777751-b04f2480-6776-11ea-8d85-45471be24df3.png)

As you can see the type definition being sent to magenta is a drastic shift away from the vscode theme. Here is the same with 'type' set to red. Still not a 100% representation of the theme but closer then the above

![image](https://user-images.githubusercontent.com/1653499/76777902-f0160c00-6776-11ea-80f3-3e10ab088892.png)

Reply:
The problem comes with things like this one:
![image](https://user-images.githubusercontent.com/17773218/76785283-2ce80000-6783-11ea-9730-53a64a69ffcb.png)

If we changed the theme as you say, both the const and the var words would be red in JS. The problem is also that the keywords are magenta. The problem you mention is also present in Python, and it's unfortunately a pretty common problem.

A good idea for a package would be a package to extend the syntax highlighting of some modes. There's some packages that do that like js2-mode or web-mode for example.

Reply:
Is that screenshot from emacs? My comparison with my changes in does not make any changes to what I see within emacs (I'm using Doom and have the javascript module enabled and using the js2-mode). In fact my suggestion of the change on type does not influence that code snippet at all. 

![image](https://user-images.githubusercontent.com/1653499/76792103-404d9800-6790-11ea-91d6-f3e662ea5f65.png)

I feel like outside of some narrow places this is a hard theme to port without some modifications. Because half of the languages don't even come close to the syntax highlighting available in VS Code that this theme leans so heavily on.

Reply:
It seems you're right. It doesn't affect Javascript. It will affect other modes such as C though.

Reply:
By the way, you can do `(setq centaur-tabs-plain-icons t)` to make the icons the same colors as the tabs' text, which will make it look more similar to the original.